support/oidc-discovery-provider/handler.go

```diff
@@ -34,14 +44,31 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if _, ok := h.domains[r.Host]; !ok {
+		http.Error(w, "domain not allowed", http.StatusNotFound)
+		return
+	}
```

Comment:
I don't think we can do this unconditionally without breaking existing deployments who have not configured their reverse proxies to add the `X-Forwarded-Host` header.

What if we only enable this validation if the `Domains` configurable was used and maintain existing behavior if `Domain` was used instead? When we finally deprecate `Domain`, we'll be in a stronger position to be doing this validation for every deployment, since those who have been using (or have migrated to) `Domains` will already have had to adjust their proxy deployment to add the header.

Reply:
Done.